src/ccache.cpp

```diff
@@ -4048,7 +4048,7 @@ int
 ccache_main(int argc, char* argv[])
 {
   try {
-    // TODO ptr, and LEAK to be available in atexit()
+    // TODO! ptr, and LEAK to be available in atexit()
     Context* ctx = new Context{};
```

Comment:
Because `Context` is passed as a `void* context` to `atexit`/`exit_fn`-called-functions it can't be a stack variable. To prevent leaks a final exit function is required to to `delete` it, though maybe only on valgrind runs.

Reply:
Sounds reasonable to pass ownership of ctx to the last executed exit function.